### DISCUSSION TITLE
Registering Custom Relationship Vue Component

### DISCUSSION THREAD

**Discussion by SiCoUK** [statamic/cms#7188] [Q&A] [answered]
I have just started Statamic and am making a custom relationship component to link to items in my database.

With reference to the custom Vue Components: https://statamic.dev/extending/relationship-fieldtypes#customizing-the-view

Where do I register the component?
```Vue.component('TwitterRelationshipItem', require('./TwitterRelationshipItem.vue'));```

I have added resources/js/cp.js to my webpack and added the above line in there but I don't see my component.

Thanks

**Comment by jasonvarga** [comment, marked as answer]
Give https://statamic.dev/extending/control-panel a read through.

Specifically this bit:


```js
Statamic.booting(() => {
    Statamic.$components.register('password-fieldtype', Password);
});
```

**Reply by SiCoUK** [reply]
Brilliant thank you. After some messing about the key bits I had missed were all in that page.

I did fall in to an error which seems to be a mix issue "Module parse failed: Unexpected token (1:0)" which is fixed by adding ```.vue({ version: 2 })``` to the end of the new mix.js line.

```mix.js('resources/js/cp.js', 'public/vendor/app/js').vue({ version: 2 })```

As I don't know much about Mix and Statamic at the moment there may be another way around this and it may be my setup (although I'm currently using the starter kit).
